projects/fsastorefrontlib/src/cms-components/change-process/change-coverage/change-coverage.component.ts

```diff
@@ -26,40 +22,37 @@ export class ChangeCoverageComponent extends AbstractChangeProcessStepComponent
               optionalProduct => !optionalProduct.isMandatory
             );
             if (optionalProducts && !this.isSimulated(changeRequestData)) {
-              optionalProducts.map((coverage, index) => {
-                const coverageCopy = { ...coverage, index };
-                this.optionalCoverages.push(coverageCopy);
-                this.initialOptionalCoverages.push({ ...coverageCopy });
-              });
+              this.optionalCoverages = [...optionalProducts];
             }
           })
         )
         .subscribe()
     );
   }
 
-  addCoverage(coverage: any) {
-    this.optionalCoverages.forEach(potentialCoverage => {
-      if (
-        potentialCoverage.coverageProduct.code === coverage.coverageProduct.code
-      ) {
-        potentialCoverage.coverageIsIncluded = true;
-      }
-    });
+  resolveCoverageChange(index: number) {
```

Comment:
Method for determining whether coverage has been changed or not. Based on that `checkCoverageChanged` array is filled or emptied. If array is empty - nothing has been changed, and the user cannot continue the process.